substrate/primitives/keyring/src/ed25519.rs

```diff
@@ -153,40 +145,51 @@ impl From<Keyring> for Pair {
 	}
 }
 
-impl From<Keyring> for [u8; 32] {
-	fn from(k: Keyring) -> Self {
-		*(*PUBLIC_KEYS).get(&k).unwrap().as_array_ref()
-	}
+macro_rules! hex2arr {
+	($input:expr) => {{
+		const PUBLIC_BYTES: [u8; 32] = hex2arr($input);
+		PUBLIC_BYTES
+	}};
 }
```

Comment:
You could make this macro generic over size:

```suggestion
macro_rules! hex2arr {
	($input:expr) => {{
		const PUBLIC_BYTES: [u8; $input.len()] = hex2arr($input);
		PUBLIC_BYTES
	}};
}
```

...and then you could export it from `const_hex2array.rs`, and hide the `hex2array` function (can't make it non-public because of the macro, but can be made `#[doc(hidden)]`) so that no one accidentally calls it in a non-const context.

Reply:
done in: 96c3574